pkg/internal/sqlsmith/scalar.go

```diff
@@ -373,6 +379,76 @@ func makeFunc(s *scope, ctx Context, typ types.T, refs colRefs) (tree.TypedExpr,
 	), typ), true
 }
 
+const maxWindowFrameOffset = 10
+
+func makeWindowFrame() *tree.WindowFrame {
```

Comment:
pass a `s *scope` here and use `s.schema.rnd.Intn` instead of the `rand.Intn` uses.